src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -1269,10 +1274,11 @@ struct Gcx
             __gshared atforkHandlersInstalled = false;
             if (!atforkHandlersInstalled)
             {
-                pthread_atfork(
-                    &_d_gcx_atfork_prepare,
-                    &_d_gcx_atfork_parent,
-                    &_d_gcx_atfork_child);
+                version (COLLECT_PARALLEL)
```

Comment:
Why are the fork handlers only needed when marking in parallel?

Reply:
You tell me. `_d_gcx_atfork_prepare` etc. are inside the `version (COLLECT_PARALLEL):` section.

Reply:
Initially the fork handler only dealt with variables for parallel marking, but https://github.com/dlang/druntime/pull/2817 made it more general, so should have moved it out of the version.
It currently doesn't make much difference, though, as version COLLECT_PARALLEL is available unconditionally ATM.

Reply:
Oh, I see. So, I screwed up then by adding these handlers in the wrong place.

Also I had completely forgotten that it was me who added them. :)

I will move them out.

`COLLECT_PARALLEL` is unconditional in that it is always turned on at the top of the file, but the point of such switches is to toggle them during debugging, so I think turning it off in this way ought to either work or the switch should just be removed (and the behavior controlled through the runtime setting).

Reply:
> I will move them out.

Done.